exchanges/account/account_types.go

```diff
@@ -34,7 +34,11 @@ type Accounts struct {
 	// TODO: Credential tracker to match to keys that are managed and return
 	// pointer.
 	// TODO: Have different cred struct for centralized verse DEFI exchanges.
-	SubAccounts map[Credentials]map[key.SubAccountCurrencyAsset]*ProtectedBalance
+	subAccounts map[Credentials]map[key.SubAccountAsset]subAccountAsset
+}
+
+type subAccountAsset struct {
```

Comment:
I don't really understand the need for change to the `key` and `subAccountAsset`. I have a [diff here](https://gist.github.com/gloriousCode/9c45dadaeb7cdc9dda728aba7c5206eb) which simply builds a new map during the run to know which assets to clear after an `Update`

Reply:
While it's possible to achieve the goal without changing the `key` and `subAccountAsset`, the original `SubAccountCurrencyAsset` structure flattens all assets and currencies. This requires processing all asset types during each update, which introduces unnecessary overhead. Since both `Update` and `GetHoldings` operations are performed on individual asset types, I believe the flattened structure is not the best approach.

Reply:
That explanation makes sense to me too 🙂